gdal/ogr/ogrsf_frmts/shape/dbfopen.c

```diff
@@ -2227,7 +2235,9 @@ DBFAlterFieldDefn( DBFHandle psDBF, int iField, const char * pszFieldName,
 
             /* load record */
             psDBF->sHooks.FSeek( psDBF->fp, nRecordOffset, 0 );
-            psDBF->sHooks.FRead( pszRecord, nOldRecordLength, 1, psDBF->fp );
+            if (psDBF->sHooks.FRead( pszRecord, nOldRecordLength, 1, psDBF->fp )!= 1) {
+              return FALSE;
```

Comment:
memleak of several things

Reply:
Ah the joys of C.  I read right past all those issues.

Reply:
Yeah, I've added some ad-hoc error handling, please let me know if there's a better approach.